devlib/module/thermal.py

```diff
@@ -0,0 +1,108 @@
+#    Copyright 2015 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import re
+
+from devlib.module import Module
+
+class TripPoint(object):
+    def __init__(self, zone, _id):
+        self._id = _id
+        self.zone = zone
+        self.temp_node = 'trip_point_' + _id + '_temp'
+        self.type_node = 'trip_point_' + _id + '_type'
+
+    def get_temperature(self):
+        """Returns the currently configured temperature of the trip point"""
+        temp_file = self.zone.target.path.join(self.zone.path, self.temp_node)
+        return self.zone.target.read_value(temp_file)
+
+    def set_temperature(self, temperature):
+        temp_file = self.zone.target.path.join(self.zone.path, self.temp_node)
+        self.zone.target_write_value(temp_file, temperature)
```

Comment:
I assume this should be

    self.zone.target.write_value(temp_file, temperature)

`self.zone.target` gets used a lot here; might be cleaner to expose it as `self.target` via a property:

    @property:
    def target(self):
         return self.zone.target